src/useSpeech/index.ts

```diff
@@ -0,0 +1,88 @@
+import { useMemo, useState } from 'react';
+import { useDeepCompareEffect } from '../useDeepCompareEffect';
+
+type SpeechOptions = {
```

Comment:
btw, i think it is better to define this type like below:
```ts
SpeechOptions = Partial<Pick<SpeechSynthesisUtterance>, 'lang' | 'pitch' | 'rate' | 'voice' | 'volume'>
```

This way it will always be compatible with types of utterance, because your variant does not support `null` for voice, although it is a valid value

Reply:
@xobotyi  Addressed [here](https://github.com/react-hookz/web/pull/1378/commits/07b0e0ed351002941a90fdd4d71e58468623877d)